src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,469 @@
+
+                foreach (KeyValuePair<ITypeSymbol, Location> pair in _targetTypes.TypesForConfigureMethodGen)
+                {
+                    ITypeSymbol type = pair.Key;
+                    string typeDisplayString = type.ToDisplayString();
+
+                    _writer.WriteBlockStart($"if (typeof(T) == typeof({typeDisplayString}))");
+
+                    _writer.WriteLine($@"return {LiteralFor_services}.{LiteralFor_Configure}<{typeDisplayString}>({LiteralFor_obj} =>");
+
+                    // Action method body.
+                    _writer.WriteBlockStart("{");
+                    TryEmitBindLogicFromIConfiguration(type, LiteralFor_obj, InitializationKind.None);
+                    _writer.WriteBlockEnd(");");
+
+                    _writer.WriteBlockEnd();
+                }
+
+                Emit_NotSupportedException_UnableToBindType(NotSupportedReason_TypeNotDetectedAsInput);
+                _writer.WriteBlockEnd();
+            }
+
+            private void EmitGetMethod()
+            {
+                _writer.WriteBlockStart($"public static T? {LiteralFor_Get}<T>(this {GlobalNameForIConfigurationType} {LiteralFor_configuration})");
+
+                foreach (KeyValuePair<ITypeSymbol, Location> pair in _targetTypes.TypesForGetMethodGen)
+                {
+                    ITypeSymbol type = pair.Key;
+                    string typeDisplayString = type.ToDisplayString();
+
+                    _writer.WriteBlockStart($"if (typeof(T) == typeof({typeDisplayString}))");
+
+                    if (UseStringBasedParse(type))
+                    {
+                        Emit_NotSupportedException_UnableToBindType(type, NotSupportedReadon_OnlyComplexObjectsSupported);
+                    }
+                    else if (TryEmitBindLogicFromIConfiguration(type, LiteralFor_obj, InitializationKind.Declaration))
+                    {
+                        _writer.WriteLine($"return (T)(object){LiteralFor_obj};");
+                    }
+
+                    _writer.WriteBlockEnd();
+                }
+
+                Emit_NotSupportedException_UnableToBindType(NotSupportedReason_TypeNotDetectedAsInput);
+                _writer.WriteBlockEnd();
+            }
+
+            private void EmitBindMethods()
+            {
+                foreach (KeyValuePair<ITypeSymbol, Location> pair in _targetTypes.TypesForBindMethodGen)
+                {
+                    EmitBindMethod("internal", pair.Key);
+                }
+
+                while (_privateBindMethodGen_QueuedTypes.Count > 0)
+                {
+                    EmitBindMethod("private", _privateBindMethodGen_QueuedTypes.Dequeue());
+                }
+            }
+
+            private void EmitBindMethod(string visibility, ITypeSymbol type)
```

Comment:
>EmitBindMethod [](http://example.com/codeflow?start=25&length=14)

How we bind to IReadOnly interfaces in this logic?

Reply:
I'll peel IReadOnly interfaces off from this PR. The goal would be to enable the rest of the collection tests.